### PR TITLE
DAOS-15605 vos: Add version param to pool create

### PR DESCRIPTION
**Pull request by liw**
In a DAOS pool using the old pool global version, we need to create new VOS pools using the old DF version. See the Jira ticket for the details. This patch adds a version parameter to vos_pool_create and vos_pool_create_ex.

If a pool with an old layout version is served by a DAOS version with a new default layout version, for instance, a 2.4-layout pool served by DAOS 2.5, then any new VOS pools created for this DAOS pool must use the old layout, or downgrading back to the old DAOS version would become impossible.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by liw** [comment]
daos_test/rebuild_simple: DAOS-15290

**Review by wangshilong — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]

**Comment by liw** [comment]
rebuild_simple: DAOS-15431

**Review by jolivier23 — APPROVED** [review]
so new pools by default will always use the old version unless the user explicitly does a pool upgrade ?  That seems a reasonable behavior.

**Comment by liw** [comment]
Merged master to resolve the conflict with the `ddb` changes.

File: src/vos/tests/vts_pool.c [hunk 1]
```diff
@@ -497,6 +490,36 @@ pool_open_excl_test(void **state)
...
+	FAULT_INJECTION_REQUIRED();
+
+	uuid_generate(uuid);
+
+	/* DF version too old. */
+	ret = vos_pool_create(arg->fname[0], uuid, VPOOL_256M, 0, 0, 1 /* version */, NULL);
+	assert_rc_equal(ret, -DER_INVAL);
+
+	/* DF version old but supported. */
+	ret = vos_pool_create(arg->fname[0], uuid, VPOOL_256M, 0, 0, VOS_POOL_DF_2_4, NULL);
```
Comment (kccain) [inline src/vos/tests/vts_pool.c:510]: No change requested. I suppose a future concern would be changing this line of code to specify whatever is the lowest supported durable format version value, for example after VOS_POOL_DF_2_4 might no longer be supported by future daos software versions.

There is a POOL_DF_VER_1 defined in vos_layout.h - could it be used here to specify as the version argument (referring always to the lowest supported version value)?

**Review by kccain — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]

**Review by wangshilong — APPROVED** [review]

File: src/mgmt/srv_target.c [hunk 2]
```diff
@@ -734,8 +734,8 @@ tgt_vos_create_one(void *varg)
 	if (rc)
 		return rc;
 
-	rc = vos_pool_create(path, (unsigned char *)vpa->vpa_uuid,
-			     vpa->vpa_scm_size, vpa->vpa_nvme_size, 0, NULL);
+	rc = vos_pool_create(path, (unsigned char *)vpa->vpa_uuid, vpa->vpa_scm_size,
```
[the remark this answers is not in the document]
Comment (liw) [reply src/mgmt/srv_target.c:737]: @kccain, yes, I think so.

File: src/vos/tests/vts_pool.c (hunk 1 above)
Comment (liw) [reply src/vos/tests/vts_pool.c:510]: @kccain, I thought about this and didn't bother to export `POOL_DF_VER_1` to the test, for when `vos` drops `VOS_POOL_DF_2_4` this test will fail to compile (or fail to pass). I can look into exporting `POOL_DF_VER_1` in a separate PR.

**Review by gnailzenh — APPROVED** [review]

**Pull request merged**